test/powershell/Modules/Microsoft.PowerShell.Utility/Export-Csv.Tests.ps1

```diff
@@ -1,73 +1,69 @@
 Describe "Export-Csv" -Tags "CI" {
-    $testObject = @("test","object","array")
-    $testCsv = Join-Path -Path $TestDrive -ChildPath "output.csv"
+    BeforeAll {
+        $testObject = @("test","object","array")
+        $testCsv = Join-Path -Path $TestDrive -ChildPath "output.csv"
+    }
 
     AfterEach {
-	Remove-Item $testCsv -Force -ErrorAction SilentlyContinue
+        Remove-Item $testCsv -Force -ErrorAction SilentlyContinue
     }
 
     It "Should be able to be called without error" {
-	{ $testObject | Export-Csv $testCsv } | Should Not Throw
+        { $testObject | Export-Csv $testCsv -ErrorAction Stop } | Should Not Throw
     }
 
     It "Should throw if an output file isn't specified" {
-	{ $testObject | Export-Csv -ErrorAction SilentlyContinue } | Should Throw
+        { $testObject | Export-Csv -ErrorAction Stop } | ShouldBeErrorId "CannotSpecifyPathAndLiteralPath,Microsoft.PowerShell.Commands.ExportCsvCommand"
     }
 
     It "Should be a string when exporting via pipe" {
-	$testObject | Export-Csv $testCsv
+        $testObject | Export-Csv $testCsv
 
-	$piped = Get-Content $testCsv
+        $piped = Get-Content $testCsv
 
-	$piped[0] | Should Match ".String"
+        $piped[0] | Should BeExactly "#TYPE System.String"
     }
 
     It "Should be an object when exporting via the inputObject switch" {
-	Export-Csv -InputObject $testObject -Path $testCsv
+        Export-Csv -InputObject $testObject -Path $testCsv
 
-	$switch = Get-Content $testCsv
+        $switch = Get-Content $testCsv
 
-	$switch[0] | Should Match ".Object"
+        $switch[0] | Should BeExactly "#TYPE System.Object[]" 
     }
 
     It "Should output a csv file containing a string of all the lengths of each element when piped input is used" {
-	$testObject | Export-Csv -Path $testCsv
-
-	$first    = "`"" + $testObject[0].Length.ToString() + "`""
-	$second   = "`"" + $testObject[1].Length.ToString() + "`""
-	$third    = "`"" + $testObject[2].Length.ToString() + "`""
-	$expected = @("#TYPE System.String", "`"Length`"", $first , $second, $third)
+        $testObject | Export-Csv -Path $testCsv
 
-	for ( $i = 0; $i -lt $testCsv.Length; $i++)
-	{
-	    $(Get-Content $testCsv)[$i] | Should Be $expected[$i]
-	}
-    }
+        $first    = "`"" + $testObject[0].Length.ToString() + "`""
+        $second   = "`"" + $testObject[1].Length.ToString() + "`""
+        $third    = "`"" + $testObject[2].Length.ToString() + "`""
+        $expected = @("#TYPE System.String", "`"Length`"", $first , $second, $third)
 
-    It "Should be able to use the epcsv alias without error" {
-	{ $testObject | Export-Csv -Path $testCsv } | Should Not Throw
+        for ( $i = 0; $i -lt $testCsv.Length; $i++) {
```

Comment:
I wonder why we use `$testCsv.Length`?
I'd expect `$expected.Count`.
Your thoughts?

Reply:
I agree and think it should be $expected.Count. I have no clue how the length of the temporary file path string is relevant to anything in this test.

Reply:
Is there even a need for looping logic here? would calling get-content once and perform 4 explicit tests be better? It seems like creating an array and looping through it (with a `for` loop no less) is just overkill.

Reply:
I think we should stop on ` $expected.Count`